sources/packages/web/src/views/institution/designations/DesignationRequest.vue

```diff
@@ -0,0 +1,107 @@
+import {
+  useFormatters,
+  useInstitutionAuth,
+  useInstitutionState,
+  useToastMessage,
+} from "@/composables";
+import { useRouter } from "vue-router";
+import DesignationAgreementForm from "@/components/common/DesignationAgreement/DesignationAgreementForm.vue";
+import {
+  DesignationModel,
+  DesignationFormViewModes,
+  DesignationLocationsListItem,
+} from "@/components/common/DesignationAgreement/DesignationAgreementForm.models";
+import { DesignationAgreementService } from "@/services/DesignationAgreementService";
+import { SubmitDesignationAgreementDto } from "@/types/contracts/DesignationAgreementContract";
+import { InstitutionRoutesConst } from "@/constants/routes/RouteConstants";
+
+export default {
+  components: { FullPageContainer, DesignationAgreementForm },
+  setup() {
+    const router = useRouter();
+    const toastMessage = useToastMessage();
+    const formatter = useFormatters();
+    const { institutionState } = useInstitutionState();
+    const {
+      userFullName,
+      userEmail,
+      isLegalSigningAuthority,
+    } = useInstitutionAuth();
+
+    const designationModel = reactive({} as DesignationModel);
+    designationModel.institutionName = institutionState.value.operatingName;
+    designationModel.institutionType = institutionState.value.institutionType;
+    designationModel.isBCPrivate = institutionState.value.isBCPrivate;
+    designationModel.viewMode = DesignationFormViewModes.submission;
+    if (isLegalSigningAuthority) {
+      // Only populates the signing officer data
+      // if the current user is has the proper role.
+      designationModel.dynamicData = {
+        legalAuthorityName: userFullName.value,
+        legalAuthorityEmailAddress: userEmail.value,
+      };
+    }
+
+    onMounted(async () => {
+      const locations = await InstitutionService.shared.getAllInstitutionLocations();
+      designationModel.locations = locations.map(
+        location =>
+          ({
+            locationId: location.id,
+            locationName: location.name,
+            locationAddress: formatter.getFormattedAddress({
+              ...location.data.address,
+              provinceState: location.data.address.province,
+            }),
+            requestForDesignation: false,
+          } as DesignationLocationsListItem),
+      );
+    });
+
+    const submitDesignation = async (model: DesignationModel) => {
+      try {
+        await DesignationAgreementService.shared.submitDesignationAgreement({
+          dynamicData: model.dynamicData,
+          locations: model.locations.map(location => ({
+            locationId: location.locationId,
+            requestForDesignation: location.requestForDesignation,
+          })),
+        } as SubmitDesignationAgreementDto);
+        toastMessage.success("Submitted", "Designation agreement submitted.");
+        await router.push({ name: InstitutionRoutesConst.MANAGE_DESIGNATION });
```

Comment:
do we need to await here?

Reply:
Because it is the last instruction in the method we do not, It will not change the behavior.